### DISCUSSION TITLE
Conversion from asciidoc

### DISCUSSION THREAD

**Discussion by brochape** [markdoc/markdoc#253] [Q&A] [answered]
Hi everyone,

We currently have the documentation of our projects and products in asciidoc. We're currently having a look at Markdoc for one of our projects and are quite satisfied with it.

With that experience, we're looking at converting our entire documentation to Markdoc but that would currently be a relatively big effort to port everything. Are there any plans to make a porting tool that would ease the transition from one to the other? Or better, does a tool like that already exist?

Thanks!

**Comment by matv-stripe** [comment, marked as answer]
Hey @brochape !

This is an interesting question - when we first introduced Markdoc to the Stripe Docs, the existing documentation set was written in [ERB templates](https://github.com/ruby/erb). Migrating these entire doc set to Markdoc was difficult because there was a lot of code inside of the templates.

Because of this, it was not easy to automatically convert from ERB -> Markdoc. We did build conversion tooling to help with the most common patterns and iteratively improved it. Overall, I would say we were able to automatically convert 85% of an average doc and then the last 15% we did manually conversion.

For Asciidoc, I think it might be possible to build some automated conversion tooling. We do not have plans at the moment to develop a porting tool, but if you start that, we would definitely be happy to provide some insight.

From reading https://docs.asciidoctor.org/asciidoc/latest/ and looking at some parser libraries - https://github.com/asciidoctor/asciidoctor.js/ and https://github.com/asciidoctor/asciidoctor , I think a high level strategy could be
1. Parse AsciiDoc content into an AST
2. Walk that AST and build up a Markdoc AST, converting AsciiDoc nodes into Markdoc nodes
3. With a Markdoc ast, we can pass it into the Markdoc experimental formatter (https://github.com/markdoc/markdoc/blob/main/index.ts#L158) to output plain Markdoc text

**Reply by brochape** [reply]
Thanks for the answer, that's clear

**Reply by matv-stripe** [reply]
Please keep us updated! A tool like this would definitely be a great contribution to the community and we would love to provide some insight
